apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/ResponseReader.kt

```diff
@@ -78,4 +80,24 @@ interface ResponseReader {
       })
     }
   }
+
+  fun ResponseField.shouldSkip(variableValues: Map<String, Any?>): Boolean {
```

Comment:
Why do we even need `shouldSkip` ? Isn't this supposed to be handled server side? If a server returns a value for a skipped field, this is most likely against [the spec](https://spec.graphql.org/June2018/#sec-Field-Collection). Maybe we can avoid this logic client side and save a few more CPU cycles?

Reply:
Yeah, I think you are right. This function is only needed in case of normalized cache. We can remove it.

Reply:
I will leave it for now in `RealResponseReader` as it used when parse normalized cache, until we replace it with stream parses.

Reply:
Right 👍